src/overlays/actors/ovl_En_Bigpo/z_en_bigpo.c

```diff
@@ -94,161 +151,1301 @@ static DamageTable D_80B65044 = {
+        this->unk220 = CLAMP_MAX(this->unk220, 1.0f);
+    }
+}
+
+// alt update func: the revealed fires under dampe's house
+void EnBigpo_UpdateFire(Actor* thisx, GlobalContext* globalCtx) {
+    EnBigpo* this = (EnBigpo*)thisx;
+    // bilboarding the fire toward player camera?
+    this->actor.shape.rot.y = BINANG_ROT180(func_800DFCDC(ACTIVE_CAM));
+    this->actionFunc(this, globalCtx);
+}
+
+s32 EnBigpo_OverrideLimbDraw2(struct GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                              struct Actor* actor, Gfx** gfx) {
+    EnBigpo* this = (EnBigpo*)actor;
+    // not fully invisible
+    if ((!((this->mainColor.a != 0) && (limbIndex != 7))) ||
+        ((this->actionFunc == EnBigpo_BurnAwayDeath) && (this->idleTimer >= 2))) {
+        *dList = NULL;
+    }
+    return 0;
+}
+
+void EnBigpo_PostLimbDraw2(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* actor,
+                           Gfx** gfx) {
+    EnBigpo* this = (EnBigpo*)actor;
+    s8 limbByte;
+    Vec3f* v1ptr; // todo: figure out better names
+    Vec3f* v2ptr;
+    Vec3f unusedVec;
+    s32 i;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    if ((this->actionFunc == EnBigpo_BurnAwayDeath) && (this->idleTimer >= 2) && (limbIndex == 8)) {
+        gSPMatrix((*gfx)++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList((*gfx)++, &D_060058B8);
+    }
+
+    if (limbIndex == 7) {
+        // we scale the vec3f... then do nothing with it?
+        SysMatrix_GetStateTranslationAndScaledY(1400.0f, &unusedVec);
+        if ((this->actionFunc == EnBigpo_BurnAwayDeath) && (this->idleTimer > 18)) {
+            if (actor->scale.x != 0.0f) {
+                Matrix_Scale(0.014f / actor->scale.x, 0.014f / actor->scale.x, 0.014f / actor->scale.x, 1);
+            }
+        }
+        SysMatrix_CopyCurrentState(&this->drawMtxF);
+    }
+
+    limbByte = D_80B65078[limbIndex];
+    if (limbByte != -1) {
+        if (limbByte < 3) {
+            SysMatrix_GetStateTranslation(&this->limbPos[limbByte]);
+            return;
+        }
+        if (limbByte == 3) {
+            SysMatrix_GetStateTranslationAndScaledX(3000.0f, &this->limbPos[limbByte]);
+            return;
+        }
+        if (limbByte == 4) {
+            SysMatrix_GetStateTranslationAndScaledY(-2000.0f, &this->limbPos[limbByte]);
+            return;
+        }
+
+        v2ptr = &this->limbPos[limbByte + 1];
+        v1ptr = D_80B65084;
+        SysMatrix_GetStateTranslationAndScaledX(-4000.0f, &this->limbPos[limbByte]);
+
+        for (i = limbByte + 1; i < ARRAY_COUNT(this->limbPos); i++) {
+            SysMatrix_MultiplyVector3fByState(v1ptr, v2ptr);
+            v2ptr++;
+            v1ptr++;
+        }
+    }
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
+
+// one of the two main draws
+void EnBigpo_Draw1(Actor* thisx, GlobalContext* globalCtx) {
```

Comment:
Which part of the poe is this drawing, or what state is it drawing for?

Reply:
I don't know, both are assigned as the main draw for regular Bigpo at different places, and one calls the other one as part of its operation.

I can't tell why there are two and are separate, so I left them numbered until I can figure out how to decomp objects.